torch/csrc/jit/runtime/profiling_record.h

```diff
@@ -8,12 +8,66 @@
 #include <torch/csrc/jit/ir/ir.h>
 
 #include <list>
+#include <map>
 #include <vector>
 
 namespace torch {
 namespace jit {
 
 using ::c10::TensorTypePtr;
+using Dimension = int64_t;
+
+struct ProfilingRecord;
+struct ShapeSymbolSets {
+  void reset() {
+    sets_.clear();
+  };
+  std::map<c10::ShapeSymbol, std::map<Dimension, c10::ShapeSymbol>> sets_;
```

Comment:
I don't understand what this stores, it is a nested map of ShapeSymbol to Dimension to Shape symbol?

Reply:
I added a comment.
We are implementing the algorithm we discussed but the implementation merges by every dimension location, not by a whole set at once. It makes the implementation so much simpler, but it's a bit of a break from the conceptual algorithm we discussed. 
```
// A helper structure used for partitioning
// a set associated with ShapeSymbol into
// subsets each associated with a unique
// Dimension (dimension value)
// if the same symbol is
// assigned to multiple `Dimension` in one of the
// profiling runs the set of dimension locations
// associated with the set will be split into
// two subsets each associated with the new Dimension
// value 
struct ShapeSymbolSets {
```